lib/Sema/CSSolver.cpp

```diff
@@ -1645,7 +1645,7 @@ bool ConstraintSystem::solveRec(SmallVectorImpl<Solution> &solutions,
 static bool shortCircuitDisjunctionAt(Constraint *constraint,
                                       Constraint *successfulConstraint,
                                       ASTContext &ctx) {
-
+  return false;
```

Comment:
I haven't been able to reproduce that slow down again, but if I could...I would have no idea why.